### PR TITLE
Add Homepage

### PR DESCRIPTION
**Pull request by varun13169**
Add Homepage

### CODE REVIEW

File: pages/homePage/homePage.css [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+.page-footer {
+    display: none;
+}
+
+.main-content {
+    width: 100%;
+    height: 100%;
+    display: flex;
+    /* align-items: center; */
+    /* justify-content: center; */
```
Comment (Omkar-Ghate) [inline pages/homePage/homePage.css]: remove the commented code if it's not for future use

**Review by Omkar-Ghate — APPROVED** [review]
Apart from the mentioned change, everything else looks good for me.

File: pages/homePage/homePage.css [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+.promo-catagory>img {
+    width: 100%;
+    height: auto;
+    transition: all .2s ease-in-out;
+}
+
+.promo-catagory>img:hover {
+    transform: scale(1.1);
+    opacity: 50%;
+
```
Comment (zoyatanweer) [inline pages/homePage/homePage.css]: remove extra spaces

File: pages/homePage/homePage.html [hunk 3]
```diff
@@ -0,0 +1,160 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Home Page</title>
+    <link rel="stylesheet" href="">
+    <link rel="stylesheet" href="./homePage.css">
```
Comment (zoyatanweer) [inline pages/homePage/homePage.html:10]: you can import your homepage.css into your styles.css file.

**Review by zoyatanweer — APPROVED** [review]
looks good to merge with suggested changes.

File: pages/homePage/homePage.css [hunk 4]
```diff
@@ -0,0 +1,204 @@
+@import "/styles.css";
+
+.page-wrap {
+
+    grid-template-areas:
+        'p_nav p_nav p_nav p_nav p_nav p_nav'
+        'p_content p_content p_content p_content p_content p_content'
+        'p_content p_content p_content p_content p_content p_content'
+    ;
```
Comment (iRohitGaur) [inline pages/homePage/homePage.css:9]: No need for repetition. You can do:

"p_nav"
"p_content"

File: pages/homePage/homePage.css (hunk 1 above)
Comment (iRohitGaur) [reply pages/homePage/homePage.css]: +1

File: pages/homePage/homePage.css [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+    transition: all .2s ease-in-out;
+}
+
+.promo-catagory>img:hover {
+    transform: scale(1.1);
+    opacity: 50%;
+
+}
+
+.promo-text {}
```
Comment (iRohitGaur) [inline pages/homePage/homePage.css]: empty class. Please remove

File: pages/homePage/homePage.css [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+    height: 100%;
+    display: flex;
+    /* align-items: center; */
+    /* justify-content: center; */
+    flex-direction: column;
+
+}
+
+.main-content__header {
+    /* background-color: chartreuse; */
```
Comment (zoef-yt) [inline pages/homePage/homePage.css]: remove the commented code

File: pages/homePage/homePage.css [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+    height: 100%;
+}
+
+.promo-catagory-holder {
+    display: flex;
+    width: 100%;
+    height: 50%;
+    gap: 2rem;
+    /* background-color: chocolate; */
```
Comment (zoef-yt) [inline pages/homePage/homePage.css]: remove the commented code

File: pages/homePage/homePage.css [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+    padding: 0.8rem
+}
+
+.promo-catagory {
+    height: 100%;
+    width: auto;
+    aspect-ratio: 1/1;
+    /* background-color: blue; */
```
Comment (zoef-yt) [inline pages/homePage/homePage.css]: remove the commented code

File: pages/homePage/homePage.css [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+/* Position the "next button" to the right */
+.next {
+    right: 0;
+    border-radius: 3px 0 0 3px;
+}
+
+/* On hover, add a black background color with a little bit see-through */
+.prev:hover,
+.next:hover {
+    background-color: rgba(0, 0, 0, 0.8);
```
Comment (zoef-yt) [inline pages/homePage/homePage.css:118]: Use var for colour

File: pages/homePage/homePage.css [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+    color: white;
```
Comment (zoef-yt) [inline pages/homePage/homePage.css]: Use var for colour

File: pages/homePage/homePage.css [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+}
+
+/* Caption text */
+.text {
+    color: #f2f2f2;
```
Comment (zoef-yt) [inline pages/homePage/homePage.css]: Use var for colour

File: pages/homePage/homePage.css [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+    padding: 0.8rem 1.2rem;
+    position: absolute;
+    bottom: 0.8rem;
+    width: 100%;
+    text-align: center;
+}
+
+/* Number text (1/3 etc) */
+.numbertext {
+    color: #f2f2f2;
```
Comment (zoef-yt) [inline pages/homePage/homePage.css]: Use var for colour

File: pages/homePage/homePage.css [hunk 13]
```diff
@@ -0,0 +1,204 @@
...
+    bottom: 0rem;
+    width: 100%;
+}
+
+/* The dots/bullets/indicators */
+.dot {
+    cursor: pointer;
+    height: 1.5rem;
+    width: 1.5rem;
+    margin: 0 2px;
```
Comment (zoef-yt) [inline pages/homePage/homePage.css]: I saw you use units for 0 too so add it here also to stay consistent

File: pages/homePage/homePage.css [hunk 14]
```diff
@@ -0,0 +1,204 @@
...
+    background-color: #bbb;
+    border-radius: 50%;
+    display: inline-block;
+    transition: background-color 0.6s ease;
+}
+
+.active,
+.dot:hover {
+    background-color: #717171;
```
Comment (zoef-yt) [inline pages/homePage/homePage.css]: Use var for colour

File: pages/homePage/homePage.html [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+                        <!-- Next and previous buttons -->
+                        <a class="prev" onclick="plusSlides(-1)">&#10094;</a>
+                        <a class="next" onclick="plusSlides(1)">&#10095;</a>
+
+                        <br>
+                        <!-- The dots/circles -->
+                        <div class="carousal_navigator" style="text-align:center">
+                            <span class="dot" onclick="currentSlide(1)"></span>
+                            <span class="dot" onclick="currentSlide(2)"></span>
+                            <span class="dot" onclick="currentSlide(3)"></span>
```
Comment (zoef-yt) [inline pages/homePage/homePage.html]: Don't use on click do this in js using id's

**Review by zoef-yt — CHANGES_REQUESTED** [review]

File: pages/homePage/homePage.html (hunk 15 above)
Comment (iRohitGaur) [reply pages/homePage/homePage.html]: +1

File: pages/homePage/homePage.css (hunk 1 above)
Comment (varun13169) [reply pages/homePage/homePage.css]: Done

File: pages/homePage/homePage.html (hunk 3 above)
Comment (varun13169) [reply pages/homePage/homePage.html:10]: My design is to drive page-specific css from styles.css, to have more flexible page structure.

File: pages/homePage/homePage.css (hunk 5 above)
Comment (varun13169) [reply pages/homePage/homePage.css]: Removed, Done.

File: pages/homePage/homePage.css (hunk 7 above)
Comment (varun13169) [reply pages/homePage/homePage.css]: Done.

File: pages/homePage/homePage.css (hunk 11 above)
Comment (varun13169) [reply pages/homePage/homePage.css]: Done.

File: pages/homePage/homePage.css (hunk 12 above)
Comment (varun13169) [reply pages/homePage/homePage.css]: Done.

File: pages/homePage/homePage.css (hunk 13 above)
Comment (varun13169) [reply pages/homePage/homePage.css]: Done.

File: pages/homePage/homePage.html (hunk 15 above)
Comment (varun13169) [reply pages/homePage/homePage.html]: Done.

File: pages/homePage/homePage.css (hunk 2 above)
Comment (varun13169) [reply pages/homePage/homePage.css]: Done.

File: pages/homePage/homePage.css (hunk 6 above)
Comment (varun13169) [reply pages/homePage/homePage.css]: Done.

File: pages/homePage/homePage.css (hunk 8 above)
Comment (varun13169) [reply pages/homePage/homePage.css]: Done.

File: pages/homePage/homePage.css (hunk 9 above)
Comment (varun13169) [reply pages/homePage/homePage.css:118]: Used var on other places using colors and background colors.
Fo this 0.8 being used here itself, hence omitting the use of variable for this.

File: pages/homePage/homePage.css (hunk 14 above)
Comment (varun13169) [reply pages/homePage/homePage.css]: Done.

File: pages/homePage/homePage.css (hunk 10 above)
Comment (varun13169) [reply pages/homePage/homePage.css]: Answered here.

https://github.com/varun13169/gardenkart/pull/9#discussion_r806411080

File: pages/homePage/homePage.css (hunk 4 above)
Comment (varun13169) [reply pages/homePage/homePage.css:9]: The suggested format will work with 1 col and 2 rows.

For now, I am keeping 6 columns for consistency with other pages and to tinker around, same tinkering is the reason for two p_content rows.

**Review by iRohitGaur — APPROVED** [review]
Good to merge 👍🏻

File: pages/homePage/homePage.css (hunk 9 above)
Comment (zoef-yt) [reply pages/homePage/homePage.css:118]: Okay cool

**Review by zoef-yt — APPROVED** [review]
The code looks good now. Good to merge.

**Comment by zoef-yt** [comment]
Please do add a better description to the PR.

File: pages/homePage/homePage.css [hunk 16]
```diff
@@ -0,0 +1,197 @@
...
+.main-content__header {
+    height: 100%;
+}
+
+.promo-catagory-holder {
+    display: flex;
+    width: 100%;
+    height: 50%;
+    gap: 2rem;
+    padding: 0.8rem
```
Comment (zoyatanweer) [inline pages/homePage/homePage.css:47]: this value can be in pixels.

File: pages/homePage/homePage.css [hunk 17]
```diff
@@ -0,0 +1,197 @@
...
+    color: var(--dui-light-theme-txt);
+}
+
+/* Caption text */
+.slideshow-container__slide-caption {
+    color: var(--dui-primary-color-p2-lgt-35pct);
+    font-size: 1.5rem;
+    padding: 0.8rem 1.2rem;
+    position: absolute;
+    bottom: 0.8rem;
```
Comment (zoyatanweer) [inline pages/homePage/homePage.css:128]: this value can be in px too

File: pages/homePage/homePage.html [hunk 18]
```diff
@@ -0,0 +1,160 @@
...
+                            <div class="slideshow-container__slide-caption">Caption Three</div>
+                        </div>
+
+                        <!-- Next and previous buttons -->
+                        <a class="slideshow-container__action-prev dui-util-txt-sn dui-util-fw-bld">&#10094;</a>
+                        <a class="slideshow-container__action-next dui-util-txt-sn dui-util-fw-bld">&#10095;</a>
+
+                        <br>
+                        <!-- The dots/circles -->
+                        <div class="slideshow-container__navigator-container" style="text-align:center">
```
Comment (zoyatanweer) [inline pages/homePage/homePage.html]: is it okay to use inline styling?

**Review by zoyatanweer — APPROVED** [review]
good to merge with suggested changes.

File: pages/homePage/homePage.css (hunk 16 above)
Comment (varun13169) [reply pages/homePage/homePage.css:47]: My functionality as of now requires it to be rem.

File: pages/homePage/homePage.css (hunk 17 above)
Comment (varun13169) [reply pages/homePage/homePage.css:128]: My functionality as of now requires it to be rem

File: pages/homePage/homePage.html (hunk 18 above)
Comment (varun13169) [reply pages/homePage/homePage.html]: Yup, cleaned it.

**Pull request merged**